src/plot_scalar.cpp

```diff
@@ -148,59 +156,107 @@ void PlotScalar::draw(wxAutoBufferedPaintDC&  dc)
     if (!m_mini)
         m_rGrid = m_rGrid.Deflate(PLOT_BORDER + (XLEFT_OFFSET/2), (PLOT_BORDER + (YBOTTOM_OFFSET/2)));
 
-    //printf("h %d w %d\n", m_rCtrl.GetWidth(), m_rCtrl.GetHeight());
-    //printf("h %d w %d\n", m_rGrid.GetWidth(), m_rGrid.GetHeight());
-
     // black background
-
-    dc.Clear();
+    int plotX = 0, plotY = 0, plotWidth = 0, plotHeight = 0;
     if (m_mini)
-        m_rPlot = wxRect(0, 0, m_rGrid.GetWidth(), m_rGrid.GetHeight());        
+    {
+        plotX = 0;
+        plotY = 0;
+    }
     else
-        m_rPlot = wxRect(PLOT_BORDER + XLEFT_OFFSET, PLOT_BORDER, m_rGrid.GetWidth(), m_rGrid.GetHeight());
-   
+    {
+        plotX = PLOT_BORDER + XLEFT_OFFSET;
+        plotY = PLOT_BORDER;
+    }
+
+    plotWidth = m_rGrid.GetWidth();
+    plotHeight = m_rGrid.GetHeight();
+    
     wxBrush ltGraphBkgBrush = wxBrush(BLACK_COLOR);
-    dc.SetBrush(ltGraphBkgBrush);
-    dc.SetPen(wxPen(BLACK_COLOR, 0));
-    dc.DrawRectangle(m_rPlot);
+    ctx->SetBrush(ltGraphBkgBrush);
+    ctx->SetPen(wxPen(BLACK_COLOR, 0));
+    ctx->DrawRectangle(plotX, plotY, plotWidth, plotHeight);
 
-    index_to_px = (float)m_rGrid.GetWidth()/m_samples;
-    a_to_py = (float)m_rGrid.GetHeight()/(m_a_max - m_a_min);
+    drawGraticule(ctx);
+    
+    index_to_px = (float)plotWidth/m_samples;
+    a_to_py = (float)plotHeight/(m_a_max - m_a_min);
 
     wxPen pen;
     pen.SetColour(DARK_GREEN_COLOR);
     pen.SetWidth(1);
-    dc.SetPen(pen);
-
+    ctx->SetPen(pen);
+
+    // Compress the samples as we don't need to draw thousands of lines for
```

Comment:
@drowe67, I'm not sure the below code (using the average) is a good way of compressing the samples but it seems to look okay in the GUI. This drops the CPU usage of the "From Radio" tab from ~55-60% on my 2019 MBP to ~30%.

Reply:
Or I guess running total right now, not average. 👍